### PR TITLE
Non Playable Character and Dialog gestion

### PR DESCRIPTION
**Pull request by Saverio976**
this will add a migician on the first map

### CODE REVIEW

File: src/npc/create_npc.c [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+#include "npc.h"
+
+const char npc_json[] = "NPC JSON";
+
+const char npc_data_callback[] = "NPC DATA";
+
+static int fill_pos_rect(any_t *json, sfIntRect *rect, sfVector2f *pos)
+{
+    int *rects = NULL;
+    float *poss = NULL;
```
Comment (guillaumeAbel) [inline src/npc/create_npc.c:24]: pos_f maybe instead of poss

File: src/npc/create_npc.c [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+{
+    object_t *npc = NULL;
+    any_t *path = NULL;
+
+    path = get_from_any(json, "d", "sprite path");
+    if (path == NULL || path->type != STR) {
+        return (NULL);
+    }
+    npc = create_object(&update_npc, NULL, scene, LAYER_ENNEMY);
+    if (npc == NULL) {
```
Comment (guillaumeAbel) [inline src/npc/create_npc.c:78]: this if and the next if can be one if

File: src/npc/init_dialog.c [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    free(dialog);
+}
+
+static object_t *create_dialog_obj(scene_t *scene)
+{
+    object_t *dialog = NULL;
+    object_t *text = NULL;
+
+    dialog = create_object(&update_dialog, NULL, scene, LAYER_MENUS);
+    if (dialog == NULL) {
```
Comment (guillaumeAbel) [inline src/npc/init_dialog.c:68]: this if and the next if could be one if

File: src/npc/update/update_npc.c [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+}
+
+static bool update_rect(object_t *npc, any_t *npcjson, float dtime)
+{
+    any_t *current = NULL;
+    any_t *rectjson = NULL;
+    any_t *time = NULL;
+
+    time = get_from_any(npcjson, "d", "time");
+    current = get_from_any(npcjson, "d", "current");
```
Comment (guillaumeAbel) [inline src/npc/update/update_npc.c:62]: current unchecked

File: src/npc/add_text_dialog.c [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+#include "my_rpg.h"
+#include "npc.h"
+
+int destroy_text_dialog(void *dialog_void, scene_t *scene, window_t *win,
+    bool force_quit)
+{
+    text_dialog_t *dialog = dialog_void;
+    player_t *player = NULL;
+
+    if (dialog_void == NULL || (dialog->time < 0.5 && force_quit == false)) {
```
[the remark this answers is not in the document]
Comment (Saverio976) [reply src/npc/add_text_dialog.c:22]: xD

File: src/npc/update/update_npc.c (hunk 4 above)
Comment (Saverio976) [reply src/npc/update/update_npc.c:62]: i check it, follow it under function that use it 👁️

**Review by guillaumeAbel — APPROVED** [review]

**Pull request merged**